docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc

```diff
@@ -223,30 +232,38 @@ quarkus.oidc.discovery-enabled=false
+An advantage is that you eliminate the need for two remote calls: a remote OIDC metadata discovery call followed by another remote call to fetch the verification keys that will not be used.
+A disadvantage is that you need to know the introspection endpoint address and configure it manually.
 
-The alternative approach is to allow discovering the OIDC metadata (which is a default option) but require that only the remote JWT introspection is performed:
+The alternative approach is to allow the default option of OIDC metadata discovery but also require that only the remote JWT introspection is performed, as shown in the following example:
 
 [source, properties]
 ----
 quarkus.oidc.auth-server-url=http://localhost:8180/realms/quarkus
 quarkus.oidc.token.require-jwt-introspection-only=true
 ----
 
-An advantage of this approach is that the configuration is simple and easy to understand, while its disavantage is that a remote OIDC metadata discovery call is required to discover an introspection endpoint address (though the verification keys will also not be fetched).
+An advantage of this approach is that the configuration is simpler and easier to understand.
+A disadvantage is that a remote OIDC metadata discovery call is required to discover an introspection endpoint address, even though the verification keys will not be fetched.
 
-Note that `io.quarkus.oidc.TokenIntrospection` (a simple `jakarta.json.JsonObject` wrapper) object will be created and can be either injected or accessed as a SecurityIdentity `introspection` attribute if either JWT or opaque token has been successfully introspected.
+The `io.quarkus.oidc.TokenIntrospection`, a simple `jakarta.json.JsonObject` wrapper object, will be created.
+It can be injected or accessed as a `SecurityIdentity` `introspection` attribute, providing either the JWT or opaque token has been successfully introspected.
 
 [[token-introspection-userinfo-cache]]
-=== Token Introspection and UserInfo Cache
+=== Token introspection and `UserInfo` cache
 
-All opaque and sometimes JWT Bearer access tokens have to be remotely introspected. If `UserInfo` is also required then the same access token will be used to do a remote call to OpenID Connect Provider again. So, if `UserInfo` is required and the current access token is opaque then for every such token there will be 2 remote calls done - one to introspect it and one to get UserInfo with it, and if the token is JWT then usually only a single remote call will be needed - to get UserInfo with it.
+All opaque and sometimes JWT Bearer access tokens must be remotely introspected.
+If `UserInfo` is also required, the same access token is used in a subsequent remote call to the OIDC provider.
+So, if `UserInfo` is required, and the current access token is opaque, two remote calls are made for every such token; one remote call to introspect the token and another to get `UserInfo`.
+If the token is JWT, only a single remote call, which gets `UserInfo`, is needed.
```

Comment:
```suggestion
If the token is JWT, only a single remote call to get `UserInfo` is needed, unless it also has to be introspected.
```

I tried to imply it in the original text with a vague `usually only a single remote call will be needed` but it is not clear at all :-)